### PR TITLE
Clarify unclear checkbox event example

### PR DESCRIPTION
**Pull request by bf4**
I ended up just testing this in the browser.

Reading through the api docs, old guides, and glimmer code I wasn't exactly
sure what this example is trying to say or if it's just wrong.

### CODE REVIEW

File: guides/release/components/built-in-components.md [hunk 1]
```diff
@@ -98,11 +99,17 @@ Checkboxes support the following properties:
 Which can be bound or set as described in the previous section.
 
 
-Checkboxes are a special input type. If you want to dispatch an action on a certain [event](https://api.emberjs.com/ember/3.11/classes/Component#event-handler-methods), you will always need to define the event name in camelCase format:
+Checkboxes are a special input type.  If you want to dispatch an action on a certain [event](https://api.emberjs.com/ember/release/classes/Component#event-handler-methods), you will always need to define the event name in [Web-API format](https://developer.mozilla.org/en-US/docs/Web/API/Document/keydown_event):
+Or per the camelCase mapping in Application.customEvents in packages/@ember/-internals/views/lib/system/event_dispatcher.js
```
Comment (bf4) [inline guides/release/components/built-in-components.md]: just showing my work. should be removed before considered for merge. but I'm just not sure whether the guide is still correct or not

File: guides/release/components/built-in-components.md [hunk 2]
```diff
@@ -67,7 +67,8 @@ The following event types are supported (dasherized format):
 * `escape-press`
 * `focus-in`
 * `focus-out`
-* `key-press`
+* `key-down`
+* `key-press` ([Deprecated Web API](https://developer.mozilla.org/en-US/docs/Web/API/Document/keypress_event))
```
Comment (bf4) [inline guides/release/components/built-in-components.md]: related https://github.com/emberjs/ember.js/pull/18646

File: guides/release/components/built-in-components.md [hunk 3]
```diff
@@ -98,13 +99,23 @@ Checkboxes support the following properties:
...
 <label for="firstname">First Name</label>
-<Input @type="checkbox" @key-press={{this.updateName}} id="firstname" />
+{{!-- This does not work: uses dasherized event name --}}
+<Input @type="checkbox" @key-down={{this.updateName}} id="firstname" />
+{{!-- This does not work: uses actual event name --}}
+<Input @type="checkbox" @keydown={{this.updateName}} id="firstname" />
+{{!-- This works: uses camelCase event name--}}
+<Input @type="checkbox" @keyDown={{this.updateName}} id="firstname" />
+{{!-- This works: uses 'on' with actual event name--}}
+<Input @type="checkbox" {{on 'keydown' this.updateName}} id="firstname" />
```
Comment (jenweber) [inline guides/release/components/built-in-components.md]:
```suggestion
<Input @type="checkbox" {{on "keydown" this.updateName}} id="firstname" />
```
The default linter enforces using double quotes.

Can we move the working examples up to the top of this list?

File: guides/release/components/built-in-components.md (hunk 2 above)
Comment (jenweber) [reply guides/release/components/built-in-components.md]: Good catch!

File: guides/release/components/built-in-components.md (hunk 3 above)
Comment (bf4) [reply guides/release/components/built-in-components.md]: will do

Comment (bf4) [reply guides/release/components/built-in-components.md]: done. Lemme know if you'd like me to rebase and squash at any point.

**Comment by jenweber** [comment]
So, I tested your examples in an app locally, and then I found that `@key-down` doesn't work for text inputs either, as a matter of fact. I am merging your PR and will follow up with another to address the incorrect inputs. I am wondering if there's a bug in Ember or if the guides were just incorrect.

**Comment by bf4** [comment]
Good call

**Pull request merged**